scripts/visualize-pipeline.sh

```
@@ -0,0 +1,584 @@
+# visualize-pipeline.sh — Generate an HTML pipeline visualization from Prow JUnit XML
+#
+# Usage:
+#   ./scripts/visualize-pipeline.sh <junit_operator.xml>          # from local file
+#   ./scripts/visualize-pipeline.sh <prow-build-url>              # from Prow job URL
+#   ./scripts/visualize-pipeline.sh <build-id>                    # from build ID (uses PR 75733)
+#
+# Output: Opens pipeline.html in the default browser
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
```

Shellcheck (0.11.0)

[warning] SCRIPT_DIR appears unused. Verify use (or export if used externally).

(SC2034)